File: solutions/2038. Remove Colored Pieces if Both Neighbors are the Same Color/2038.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def winnerOfGame(self, colors: str) -> bool:
    countAlice = 0
    countBob = 0

    for a, b, c in zip(colors, colors[1:], colors[2:]):
      if 'A' == a == b == c:
        countAlice += 1
      elif 'B' == a == b == c:
        countBob += 1   

    return countAlice > countBob 

File: solutions/2038. Remove Colored Pieces if Both Neighbors are the Same Color/2038.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  bool winnerOfGame(string_view colors) {
    int countAlice = 0;
    int countBob = 0;
        
    for (int i = 1; i < colors.size() - 1; ++i)
      if (colors[i - 1] == colors[i] && colors[i] == colors[i + 1])
        if (colors[i] == 'A')
          ++countAlice;
        else
          ++countBob;
        
    return countAlice > countBob;
  }
};